pkg/reconciler/revision/resources/deploy.go

```diff
@@ -133,15 +161,28 @@ func makePodSpec(rev *v1.Revision, cfg *config.Config) (*corev1.PodSpec, error)
 	}
 
 	var extraVolumes []corev1.Volume
+
+	podInfoFeature, podInfoExists := rev.Annotations[PodInfoFeatureKey]
```

Comment:
Something to consider is that if the operator sets the feature to `Allowed` that means end-users could opt-out of annotation being mounted and potentially breaking security plugins.

Reply:
@dprotaso , I am not sure what you mean here. All plugins in my mind need to have at least one config point to indicate if they are to be used on not. I believe the normal behavior for upstream is to add Options/Extensions/Plugins that can then be used. So to use any such Option, we need to also mount. If we don't mount it is equivalent to not activating any of the options. This is why I preferred Allowed as a default, but at the same time, we can have Disabled as the default and have the upstream change it to Allowed when adding options.

Reply:
> All plugins in my mind need to have at least one config point to indicate if they are to be used on not.

I'm referring to the `PodInfoFeatureKey` not being present - thus the annotations won't be mounted in the queue proxy which to me implies no security plugins will be activated. Thus it doesn't matter what security annotations are present on the revision.

This is just an observation that we'll want clear documentation for administrators - so end-users don't have ways opt of using the plugins

Reply:
Right, 
So the documentation will say  - if you add plugins, you need to also change  the feature config map to `Allowed` or `Enabled`. If you use `Allowed` for feature to be activated the service must also include the `features.knative.dev/mount-podinfo: enable` annotation